### PR TITLE
compile cgobj.d for all builds

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/backend/cgobj.d [hunk 1]
```diff
@@ -99,10 +95,25 @@ else
...
-int obj_namestring(char *p,const(char)* name);
-
-static if (TARGET_WINDOS)
+version (Windows)
 {
+    extern(C) char* strupr(char*);
+}
+version (Posix)
+{
+    extern(C) char* strupr(char* s)
```
Comment (ibuclaw) [inline src/dmd/backend/cgobj.d:104]: Use dmd.root.port.strupr?

Comment (WalterBright) [reply src/dmd/backend/cgobj.d:104]: I keep the backend separate from the front end.

Comment (ibuclaw) [reply src/dmd/backend/cgobj.d:104]: You are already using dmd.root.hash and dmd.root.longdouble in the backend, and port is a leaf module (it only depends on druntime).  So I don't really see that as an acceptable argument.

Comment (WalterBright) [reply src/dmd/backend/cgobj.d:104]: There are other redundancies between the back and front ends, but recall the same backend is used for DMC++, and entangling it with the D frontend will cause me all sorts of headaches.

**Review by ibuclaw — APPROVED** [review]

File: src/dmd/backend/cgobj.d (hunk 1 above)
Comment (ibuclaw) [reply src/dmd/backend/cgobj.d:104]: `version (MARS)` ought to cover the dmd/dmc divide then.

**Pull request merged**